compaction/time_window_compaction_strategy.cc

```diff
@@ -225,19 +225,20 @@ time_window_compaction_strategy::get_sstables_for_compaction(table_state& table_
     std::unordered_set<shared_sstable> expired;
 
     if (db_clock::now() - _last_expired_check > _options.expired_sstable_check_frequency) {
-        clogger.debug("TWCS expired check sufficiently far in the past, checking for fully expired SSTables");
+        clogger.debug("[{}] TWCS expired check sufficiently far in the past, checking for fully expired SSTables", fmt::ptr(this));
```

Comment:
How is logging `this` help? I thought there is just one of these per table.

Reply:
IF we always use that one by reference.
The debugging code helps if the compaction strategy object is copied and the side effects in this function happen on a temporary copy rather than on the authoritative instance.